### DISCUSSION TITLE
Annotating in Rmd file

### DISCUSSION THREAD

**Discussion by jackchenry** [jokergoo/ComplexHeatmap#680] [Q&A]
Hi, this isn't really an issue so I didn't want to post it in that tab, but are you able to annotate heatmaps in a rmarkdown file? When I try to do it, I get a "can't find viewport" error. I can only get it to work through the console and plots page. I was wondering if there was a get around this or if this just is not possible.

**Comment by jokergoo** [comment]
How did you implement it? Did you use `decorate_*()` function? Maybe it is better if you can also show me your code.

**Reply by jackchenry** [reply]
I feel a little stupid, I realise that it works perfectly if I run the whole chunk. Just not line-by-line which makes sense now. Great package btw
